### PR TITLE
Add GitHub Actions deployment file

### PR DESCRIPTION
**Pull request by tjcrone**

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: deploy-book
+
+# Trigger the workflow on push to main branch
+on:
+  push:
+    branches:
+    - master
+    paths:
+    - src/**
```
Comment (rabernat) [inline .github/workflows/deploy.yml:9]: Can we also have this flow run on PRs? That's very useful for checking contributions are valid before they hit master...

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      run: |
+        pip install -r requirements.txt
+
+    # Build the book
+    - name: Build the book
+      run: |
+        jupyter-book build src
+
+    # Deploy the book's HTML to earth-env-data-science.github.io
+    - name: GitHub Pages action
```
Comment (rabernat) [inline .github/workflows/deploy.yml:42]: ... and then here you can use an if statement to only deploy on master
```suggestion
    - name: GitHub Pages action
      if: github.ref == 'refs/heads/master'
```

copied from.

https://github.com/pangeo-gallery/pangeo-gallery/blob/c038d244006bb772ee2d511391c386311f5c7ec9/.github/workflows/build-and-deploy.yml#L39

**Review by rabernat — COMMENTED** [review]
Looks great. Just one suggestion.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (tjcrone) [reply .github/workflows/deploy.yml:9]: Done.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (tjcrone) [reply .github/workflows/deploy.yml:42]: Done.

**Pull request merged**